torchaudio/transforms.py

```diff
@@ -595,6 +596,108 @@ def forward(self, waveform: Tensor) -> Tensor:
+    """
+    __constants__ = ['sample_rate', 'n_filter', 'n_lfcc', 'dct_type', 'top_db', 'log_lf']
+
+    def __init__(self,
+                 sample_rate: int = 16000,
+                 n_filter: int = 128,
+                 f_min: float = 0.,
+                 f_max: Optional[float] = None,
+                 n_lfcc: int = 40,
+                 dct_type: int = 2,
+                 norm: str = 'ortho',
+                 log_lf: bool = False,
+                 speckwargs: Optional[dict] = None) -> None:
+        super(LFCC, self).__init__()
+        supported_dct_types = [2]
+        if dct_type not in supported_dct_types:
+            raise ValueError('DCT type not supported: {}'.format(dct_type))
+        self.sample_rate = sample_rate
+        self.f_min = f_min
+        self.f_max = f_max if f_max is not None else float(sample_rate // 2)
+        self.n_filter = n_filter
+        self.n_lfcc = n_lfcc
+        self.dct_type = dct_type
+        self.norm = norm
+        self.top_db = 80.0
+        self.amplitude_to_DB = AmplitudeToDB('power', self.top_db)
+
+        if speckwargs is not None:
+            self.Spectrogram = Spectrogram(**speckwargs)
+        else:
+            self.Spectrogram = Spectrogram()
+
+        if self.n_lfcc > self.Spectrogram.n_fft:
+            raise ValueError('Cannot select more LFCC coefficients than # fft bins')
+
+        filter_mat = F.linear_fbanks(
+            n_freqs=self.Spectrogram.n_fft // 2 + 1,
+            f_min=self.f_min,
+            f_max=self.f_max,
+            n_filter=self.n_filter,
+            sample_rate=self.sample_rate,
+        )
+        self.register_buffer("filter_mat", filter_mat)
+
+        dct_mat = F.create_dct(self.n_lfcc, self.n_filter, self.norm)
+        self.register_buffer('dct_mat', dct_mat)
+        self.log_lf = log_lf
+
+    def forward(self, waveform: Tensor) -> Tensor:
+        r"""
+        Args:
+            waveform (Tensor): Tensor of audio of dimension (..., time).
+
+        Returns:
+            Tensor: Linear Frequency Cepstral Coefficients of size (..., ``n_lfcc``, time).
+        """
+        specgram = self.Spectrogram(waveform)
+
+        # adopted from mel scale
+        # pack batch
+        shape = specgram.size()
+        specgram = specgram.reshape(-1, shape[-2], shape[-1])
+
+        specgram = torch.matmul(specgram.transpose(1, 2), self.filter_mat)
+        specgram = specgram.transpose(1, 2)
+
+        # unpack batch
+        specgram = specgram.reshape(shape[:-2] + specgram.shape[-2:])
+
+        if self.log_lf:
+            log_offset = 1e-6
+            specgram = torch.log(specgram + log_offset)
+        else:
+            specgram = self.amplitude_to_DB(specgram)
+
+        # (..., channel, n_filter, time).transpose(...) dot (n_filter, n_lfcc)
+        # -> (..., channel, time, n_lfcc).transpose(...)
+        lfcc = torch.matmul(specgram.transpose(-2, -1), self.dct_mat).transpose(-2, -1)
```

Comment:
The formula can be refactored by torch.einsum, for example:
```
lfcc = torch.einsum('...cnt,nl->...ctl', specgram,self.dct_mat)
```

Reply:
Addressed in 75da309. Since my implementation is simply adapted from MFCC, I also refactored that in 04e2305. If you want we can split that off into a seperate PR.